components/theme-toggle/theme-toggle.module.css

```diff
@@ -0,0 +1,71 @@
+@keyframes toLight {
+  from {
+    transform: rotate(0deg);
+  }
+
+  to {
+    transform: rotate(180deg);
+  }
+}
+
+@keyframes toDark {
+  from {
+    transform: rotate(180deg);
+  }
+
+  to {
+    transform: rotate(360deg);
+  }
+}
+
+.iconsMask {
+  overflow: hidden;
+  border-radius: 100%;
+  width: 36px;
+  height: 36px;
+  border: 2px solid #efefef;
+  background: #ffffff11;
+  /* Safari causes overflow: hidden to not work when animating causing a 
+   * strange artifact while the icons rotate around. `isolation` is required
+   * to prevent this issue.
+  */
+  isolation: isolate;
+}
+
+.input {
+  display: none;
+}
+
+.labelDark,
+.labelLight {
+  padding: 6px;
+  height: 100%;
+  user-select: none;
+  display: flex;
+}
+
+.icons {
+  --gap-size: 12px;
+  display: flex;
+  gap: var(--gap-size);
+  height: 100%;
+  transition: transform 0.5s;
+  transform-origin: calc(100% + (var(--gap-size) / 2));
+}
+
+.icon {
+  height: 100%;
+}
+
+/* Initial state for page when light color scheme is preferred. */
+:global(.light) .icons {
+  transform: rotate(180deg);
+}
+
+:global(.light) .input:checked ~ .icons {
+  animation: toLight 0.5s linear 0s 1 forwards normal;
+}
+
+:global(.dark) .input:checked ~ .icons {
+  animation: toDark 0.5s linear 0s 1 forwards normal;
```

Comment:
Try each of these with `ease` instead of `linear` and see what you think. I think it gives the animation a little pop and feels pretty nice.

Reply:
I agree, it does look pretty smooth. I'll update it.